packages/base-controller/CHANGELOG.md

```diff
@@ -6,6 +6,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [4.0.0]
+### Added
+- Add `ControllerGetStateAction` and `ControllerStateChangeEvent` types ([#1890](https://github.com/MetaMask/core/pull/1890), [#2029](https://github.com/MetaMask/core/pull/2029))
+- Add `NamespacedName` type ([#1890](https://github.com/MetaMask/core/pull/1890))
+  - This is the narrowest supertype of all names defined within a given namespace.
+
```

Comment:
```suggestion
- Add `NotNamespacedBy` type ([#2051](https://github.com/MetaMask/core/pull/2051))

```

Reply:
Ah yes — covered in [68ac7c4](https://github.com/MetaMask/core/pull/2063/commits/68ac7c46c9776723959eb731c2dcc72710479793)

Reply:
The entry in https://github.com/MetaMask/core/pull/2063/commits/68ac7c46c9776723959eb731c2dcc72710479793 is inaccurate, since the type does not return a positive result for non-namespaced names. I think the suggestion I have here more accurately describes the type.